### ISSUE TITLE
Allow Async Function for modifyRequest

### ISSUE THREAD

**Issue by atomdmac** [zackify/react-use-upload#16]
I have a use case where I'd like to attach an `Authorization` header but will need to invoke an async function to retrieve it.  I can probably work around this by retrieving the value in advance but that is obviously not ideal.  What I'd like is to be able to provide an async function for `modifyRequest`.

With that said, from looking at the code, it seems that `modifyRequest` actually modifies the _options_ that are passed to the `request` "factory" function so it's possible that I'm just misunderstanding the intended use for `modifyRequest`.

**Comment by zackify** [comment]
Yeah... I had always planned on refactoring that part. I want to just make useUpload itself take a callback with the current request. 

But I should be able to make this change in the meantime for you. Hopefully this week.

**Comment by atomdmac** [comment]
FWIW, I think keeping `modifyRequest` where it is (i.e. as part of the "create client" factory input) works just fine (at least for my use case).  My plan is to keep my "upload client" instance in the same module as my normal HTTP Link / Apollo client code.

**Comment by zackify** [comment]
try this out, i didnt have time to test, so i threw it up as a beta release https://github.com/zackify/react-use-upload/releases/tag/0.5.13-beta

**Comment by atomdmac** [comment]
Thanks for putting this together!  

It looks like maybe a `.` is missing in this import: https://github.com/atomdmac/react-use-upload/blob/2d57e7da7dae7eb88789f400119180eda39cae8b/src/use-upload.tsx#L13.  (i.e. import { GraphQLOptions } from './clients/graphql';`

```
Cannot find module 'clients/graphql'.  TS2307

    11 | import { XHRClient, XHROptions, createXhrClient } from './clients/xhr';
    12 | import { FileOrFileList } from './';
  > 13 | import { GraphQLClient, GraphQLOptions } from 'clients/graphql';
       |                                               ^
    14 |
    15 | type HookProps = {
    16 |   files: File | FileList;
```

Git shows that this line hasn't changed since last year though so I'm not sure why it's breaking now 🤷‍♂️

**Comment by atomdmac** [comment]
Even weirder: it's breaking in the same spot on `master` for me now 🤔

**Comment by zackify** [comment]
released the next version 0.5.14

**Comment by atomdmac** [comment]
@zackify Awesome, thank you.  Looks to be working in the latest release.  Nice work 👍
